src/main/java/edu/harvard/iq/dataverse/UserServiceBean.java

```diff
@@ -34,5 +43,452 @@ public AuthenticatedUser save( AuthenticatedUser user ) {
+
+        dbRoleResults = nativeQuery.getResultList();
+        if (dbRoleResults == null){
+            return userRoleLookup;
+        }
+        
+        
+        for (Object[] dbResultRow : dbRoleResults) {            
+            
+            String groupIdentifier = UserUtil.getStringOrNull(dbResultRow[0]);
+            String groupRole = UserUtil.getStringOrNull(dbResultRow[1]);
+            if ((groupIdentifier != null)&&(groupRole != null)){  // should never be null
+                
+                List<String> groupUserList = groupsLookup.get(groupIdentifier);
+                
+                if (groupUserList != null) {
+                
+                    for (String groupUserIdentifier : groupUserList) {
+                        groupUserIdentifier = "@" + groupUserIdentifier; 
+                        //System.out.println("Group user: "+groupUserIdentifier);
+                        List<String> userRoleList = userRoleLookup.getOrDefault(groupUserIdentifier, new ArrayList<String>());
+                        if (!userRoleList.contains(groupRole)){
+                            //System.out.println("User Role: "+groupRole);
+                            userRoleList.add(groupRole);
+                            userRoleLookup.put(groupUserIdentifier, userRoleList);
+                        }
+                    }
+                }
+            }
+        }
+        
+        
+        return userRoleLookup;
+    }
+    
+    /**
+     * 
+     * @param userId
+     * @return 
+     */
+    private String getUserRolesAsString(Integer userId) {
+        String retval = "";
+        String userIdentifier = "";
+        String qstr = "select useridentifier ";
+        qstr += " FROM authenticateduser";
+        qstr += " WHERE id = " + userId.toString();
+        qstr += ";";
+        
+        Query nativeQuery = em.createNativeQuery(qstr);
+
+        userIdentifier = '@' + (String) nativeQuery.getSingleResult();
+
+        qstr = " select distinct d.name from roleassignment a, dataverserole d";
+        qstr += " where d.id = a.role_id and a.assigneeidentifier='" + userIdentifier + "'"
+                + " Order by d.name;";
+
+        nativeQuery = em.createNativeQuery(qstr);
+
+        List<Object[]> roleList = nativeQuery.getResultList();
+
+        for (Object o : roleList) {
+            if (!retval.isEmpty()) {
+                retval += ", ";
+            }
+            retval += (String) o;
+        }
+        return retval;
+    }
+    
+    /**
+     * 
+     * Run a native query, returning a List<Object[]> containing
+     * AuthenticatedUser information as well as information about the 
+     * Authenticated Provider (e.g. builtin user, etc)
+     * 
+     * @param searchTerm
+     * @param sortKey
+     * @param resultLimit
+     * @return 
+     */
+    private List<Object[]> getUserListCore(String searchTerm, String sortKey, Integer resultLimit, Integer offset) {
```

Comment:
Yes, the method may be used by an API call where all of the params can be null/unspecified.